ricecooker/classes/files.py

```diff
@@ -768,6 +774,210 @@ def process_file(self):
         return self.filename
 
 
+def denest_xml_value(value, preferred_language):
+    if isinstance(value, dict):
+        # Handle the 'string' -> '#text' nested structure
+        if "string" in value:
+            return denest_xml_value(value["string"], preferred_language)
+        elif "langstring" in value:
+            return denest_xml_value(value["langstring"], preferred_language)
+        # Handle other simple text and key-value pairs
+        elif "#text" in value:
+            return value["#text"]
+        elif "value" in value:
+            return value["value"]
+    elif isinstance(value, list):
+        try:
+            return next(
+                denest_xml_value(item, preferred_language)
+                for item in value
+                if item.get("@language", "").startswith(preferred_language)
+            )
+        except StopIteration:
+            return [denest_xml_value(item, preferred_language) for item in value]
+    return value
+
+
+class IMSCPZipFile(DownloadFile):
+    default_ext = file_formats.HTML5
+    allowed_formats = [file_formats.HTML5]
+    is_primary = True
+
+    def get_preset(self):
+        return self.preset or format_presets.IMSCP_ZIP
```

Comment:
This is producing a server error in Studio https://learningequality.sentry.io/issues/4997106816/?project=1252819